src/jit/codegenarm64.cpp

```diff
@@ -2446,46 +2450,49 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     //     (MinInt / -1) => ArithmeticException
                     //
                     bool checkDividend = true;
-                    // Do we have a contained immediate for the 'divisorOp'?
-                    if (divisorOp->isContainedIntOrIImmed())
+
+                    // Do we have an immediate for the 'divisorOp'?
+                    // 
+                    if (divisorOp->IsCnsIntOrI())
                     {
-                        GenTreeIntConCommon* intConst = divisorOp->AsIntConCommon();
-                        assert(intConst->IconValue() != 0);      // already checked above by IsZero()
-                        if (intConst->IconValue() != -1)
+                        GenTreeIntConCommon* intConstTree = divisorOp->AsIntConCommon();
+                        ssize_t intConstValue = intConstTree->IconValue();
+                        assert(intConstValue != 0);      // already checked above by IsZero()
+                        if (intConstValue != -1)
                         {                            
                             checkDividend = false;    // We statically know that the dividend is not -1
                         }
                     }
-                    else
+                    else  // insert check for divison by zero 
                     {   
                         // Check if the divisor is zero throw a DivideByZeroException
-                        emit->emitIns_R_I(INS_cmp, cmpSize, divisorReg, 0);
+                        emit->emitIns_R_I(INS_cmp, size, divisorReg, 0);
                         emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
                         genJumpToThrowHlpBlk(jmpEqual, SCK_DIV_BY_ZERO);
-
+                    }
+                    
+                    if (checkDividend)
+                    {
```

Comment:
On arm64 the hardware never generates any faults for the integer divide instruction.
On arm32 there aretwo cases:
1) the hardware faults and doesn't implement the integer divide instruction
2) it behaves as arm64 does.

The arm32 hardware that we used back when we developed the arm32 JIT  was case #1, so we had to use JITR helper calls to implement integer divide.